DOCUMENTATION.md

```diff
@@ -1370,529 +1395,202 @@ The following are example configurations.
+## 7.2. Node.js Client Operation Modes
 
-1. Using the built-in `BasicSSLOptionsFactory` bundled with the client.
-2. Writing an `SSLOptionsFactory`.
+The client has two operation modes because of the distributed nature of the data and cluster: smart and unisocket.
 
-Below subsections describe each way.
+### 7.2.1. Smart Client
 
-**Using the Built-in `BasicSSLOptionsFactory`**
+In the smart mode, the clients connect to each cluster member. Since each data partition uses the well known and consistent hashing algorithm, each client can send an operation to the relevant cluster member, which increases the overall throughput and efficiency. Smart mode is the default mode.
 
-Hazelcast Node.js client includes a utility factory class that creates the necessary `options` object out of the supplied
-properties. All you need to do is to specify your factory as `BasicSSLOptionsFactory` and provide the following options:
+### 7.2.2. Unisocket Client
 
-- `caPath`
-- `keyPath`
-- `certPath`
-- `servername`
-- `rejectUnauthorized`
-- `ciphers`
+For some cases, the clients can be required to connect to a single member instead of each member in the cluster. Firewalls, security or some custom networking issues can be the reason for these cases.
 
-See [`tls.connect`](https://nodejs.org/api/tls.html#tls_tls_connect_options_callback) of Node.js for the descriptions of each option.
+In the unisocket client mode, the client will only connect to one of the configured addresses. This single member will behave as a gateway to the other members. For any operation requested from the client, it will redirect the request to the relevant member and return the response back to the client returned from this member.
 
-> `caPath`, `keyPath` and `certPath` define the file path to the respective file that stores such information.
+## 7.3. Handling Failures
 
-```json
-{
-    "network": {
-        "ssl": {
-            "enabled": true,
-            "factory": {
-                "exportedName": "BasicSSLOptionsFactory",
-                "properties": {
-                    "caPath": "ca.pem",
-                    "keyPath": "key.pem",
-                    "certPath": "cert.pem",
-                    "rejectUnauthorized": false
-                }
-            }
-        }
-    }
-}
-```
+There are two main failure cases you should be aware of. Below sections explain these and the configurations you can perform to achieve proper behavior.
 
-If these options are not enough for your application, you may write your own options factory and instruct the client
-to get the options from it, as explained below.
+### 7.3.1. Handling Client Connection Failure
 
-**Writing an `SSLOptionsFactory`**
+While the client is trying to connect initially to one of the members in the `ClientNetworkConfig.addressList`, all the members might not be available.
+Instead of giving up, throwing an error and stopping the client, the client retries to connect as configured which is
```

Comment:
nit: change ` which is described in the...` to `. This behavior is described in the...`